src/services/env/Env.ts

```diff
@@ -58,7 +58,12 @@ export default class Env {
 
     if (pathConfigNode instanceof HTMLScriptElement) {
       try {
-        return JSON.parse(pathConfigNode.innerText.trim())
+        const config = JSON.parse(pathConfigNode.innerText.trim()) as PathConfig
+
+        // Ensures the config always has an absolute URL.
+        config.apiUrl = getAbsoluteUrl(window.location.origin, config.apiUrl)
```

Comment:
How come we are still reading `window.location.origin` shouldn't we just be using whatever is passed into the gui via the HTML vars?

Reply:
What we get from Kuma is not an absolute URL but a root-relative URL path like `'/'`. Without the concatenation with `window.location.origin`, this would result in requests made to `http:///meshes/...` which is wrong. We need this as long as we can’t guarantee that we always deal with absolute URLs.

Reply:
Sorry bear with me I'm just looking into this, I would have thought having `/` as a base path would be the same as 'on whichever host I am on' i.e. root.

It looks to me at first glance that the problem in production is that the URLs passed through to `makeRequest` look like `//policies` not `/policies`.

Reply:
Just before I have another look, if this is the case:

> It looks to me at first glance that the problem in production is that the URLs passed through to makeRequest look like `//policies` not `/policies`.

Would you do anything in the PR differently?

Reply:
As mentioned by @kleinfreund there are 2 options for the var in the index.html:

1. a complete url (containing a scheme)
2. an absolute path (but not an absolute URL).

If we're in 2 it is assumed that we're serving the api from the same url as the static files (so window.location.origin).

Reply:
Right yeah gotcha. I think where I might not be following is, it looks like if you specify an absolute path right now i.e. `/`, it makes requests to `//policies` which seems to expand to `http://policies`. If the slashes didn't get doubled up in the frontend code then the requests would be `/policies` not `//policies` and therefore just use localhost i.e. 'from the same url as the static files'?

Maybe I'm not following here?

Reply:
Requests in the production build are currently made to URLs of the form `http://policies/` which is just broken structurally.

I’m just ensuring that we always concatenate the origin correctly before making requests. Whether the URLs are explicitly constructed using `window.location.origin` or `/` doesn’t really matter (they’re the same as far as `fetch` is concerned).

Reply:
Ok well it just felt like a lot of code/complexity for what I would have thought would be just `${stripLastSlashIfThereIsOne(config.apiURL)}/policies`.

> Without the concatenation with window.location.origin, this would result in requests made to http:///meshes/... which is wrong. We need this as long as we can’t guarantee that we always deal with absolute URLs.

I guess I wanted to point out that I think the bug isn't that we aren't prepending `window.location.origin`, its that by default in prod we are using double slashes on everything.

So to maybe finish up here, it sounds like you wouldn't do anything differently here considering the above?

Reply:
Not that it matters, but I slightly simplified the implementation to not insist on making URLs absolute.

Reply:
Ok I took another brief scan, so we are not using `window.location.origin` anymore and ensuring we don't double up the slashes, and just use root `/` instead (or whatever comes from `config.apiUrl`)? If so, that sounds good, and I can take another look at it.

Reply:
I did not check we don’t use `window.location.origin` at all anymore. It is not relevant to this PR so I’d prefer it to not play a role in review.

This PR remains reviewable.